src/store/config/selectors.ts

```diff
@@ -25,3 +26,25 @@ export const getFirestoreConfigResult = createSelector(
   getConfigResult,
   result => map(result, config => config.firestore)
 );
+
+export function useProjectId() {
+  const projectId = useSelector(getProjectIdResult);
+  if (hasError(projectId)) {
+    throw new Error('ProjectId erred');
+  }
+  if (!projectId?.data) {
+    throw new Error('ProjectId is not yet defined');
```

Comment:
This is really awkward since it is an error, not a "pending" state. Thoughts? Or shall we `swr` the config and call it a day in a follow-up PR?

Reply:
Cleaned this up... added a common useRemoteResultData hook that is then called by useProjectId and useFirestoreConfig.